### DISCUSSION TITLE
Exensible Dashboard with custom widgets/portlets

### DISCUSSION THREAD

**Discussion by papyr** [mixcore/mix.core#495] [Ideas & Features]
Feature request:
### **Yes a feature request, Most CMS fall short when it comes to metrics** / displaying metrics as a dashboard
And as such making it an uphill task to adopt in an organization that may need to display metrics as well. For e.g. bay occupied, people in queue, weather, forms to print. The default has been sharepoint or crazy expensive ABP, etc.

**Describe the solution you'd like**

Since mix core provides a lot of common framework, and content support, developers could adopt and benefit their organizations, for e.g.  those that support small dev teams and mid organizations/non-profits/small shops/teams/sports etc...  It would be great if we can also _house a tabbed dashboard_, that can host multiple drag-drop widgets on the dashboard panel.

Each widget should sit in widget panel, (enhancements) should allow common functionality such as.. print, export to CSV, excel, lock, edit or even change chart type. These could mix.core components that developers can share, resell on your store etc.

**Describe alternatives you've considered**

Now slowly since many applications are trending towards this, in various domains need a snapshot view (CMS, non-profits, BackOffice internet site, eCommerce, Sharepoint, DNN) [ABP .io](https://docs.abp.io/en/abp/latest/UI/AspNetCore/Widgets) has introduced these dashboard widgets but there's a crazy non-affordable commercial angle to it, the moment a small company wants to use it.

Also please consider a pricing solution where we can buy in at a very affordable path from premium to premium

**Additional context**

1. The dashboard can house the core widgets for Mix.Core
2. It can also be extensible for users to add their own custom metrics
3. It should be able to be either public facing, or internal role based.
4. Ajax or responsive (avoid full page refresh for data updates)

![image](https://user-images.githubusercontent.com/12733414/131631123-cefdf345-edb4-4433-ada9-7016eca5f39b.png)
![image](https://user-images.githubusercontent.com/12733414/131631244-bbace798-4a71-49c7-a48b-6e4147bf2d2e.png)

![image](https://user-images.githubusercontent.com/12733414/131630902-cec25d38-e6b2-420d-8a29-bf5cca7972ad.png)

**Comment by papyr** [comment]
Hello did I label wrongly, I put it under a feature request -- with code samples.

**Reply by Smilefounder** [reply]
Hi @papyr please no worry, I just convert the issue to a discussion topic instead.

**Comment by Smilefounder** [comment]
Hi @papyr thanks a lot for your concerns, and we are totally aware of this. This is the reason why we are starting the new Angular Portal (https://github.com/mixcore/mix.portal.angular) that allows developers can do a lot of fancy things.

But will take us a while to bring this new portal release. Feel free to let us know if you have any other concerns.

**Reply by papyr** [reply]
- WOW 🎁 , but please make it easy for developer to extend customize and house new types of widgets. My boss is paying 500 for license and getting very bad support with another asp cms. I showed him this..
- Also please make 2-way-binding easy with permissions. for e.g. managers only can see salary
- Easy generation of mix.core modules/plugins
- Allow groups, for e.g. east coast managers vs. UK managers can see only their salaries. These groups should also allow vendors to login to the Mix.Core portal.


Please see the samples I added, at least the demos, they allow settings on the widgets and changing chart types.

**Reply by papyr** [reply]
https://github.com/jdanylko/Tuxboard

**Reply by Smilefounder** [reply]
Thanks @papyr for your information.

cc @nhathoang989 @nguyentuansi @phongcao3091998
